### ISSUE TITLE
Question: using vncdotool under multiprocess

### ISSUE THREAD

**Issue by guicho271828** [sibson/vncdotool#182]
- [x] VNC server and version: tightvncserver running in a ubuntu:18.04 docker container
 - [x] vncdotool version d7a98e2925ac2885ffdd6996cbd0491e3e2d6759
 - [ ] steps to reproduce 

Is it possible that using vncdotool in a subprocess spawned by `multiprocess` module causes an issue?
It seems `queue.get` in `api.py` (in `proxy_call`) times out.
The issue does not occur when I limit the number of subprocesses to be 1.

architecture sketch below
```
+ main process
   + subprocess1
      -> api.connect(server1::port1, password=XXX, timeout=5)
      -> mouseMove, etc. -> times out
   + subprocess2
      -> api.connect(server1::port2, password=XXX, timeout=5)  # connected to a different VNC server inside a docker container
      -> mouseMove, etc. -> times out
```

Disconnecting & reconnecting to the same port sometimes works, but it is very unreliable, and it eventually stops working.

**Comment by sibson** [comment]
I'm not aware of any reason this wouldn't work. Some vncservers have a setting that only allow a single connection.

**Comment by guicho271828** [comment]
Each vncserver is a separate instance in a separate docker container, no one else is connecting to them, i.e. guaranteed to be a single connection.

**Comment by guicho271828** [comment]
In other words, I have multiple vncservers running at the same time, each running in completely separate docker containers, and trying to mass-control them in parallel (using `multiprocess`) from a single python program.

**Comment by pmhahn** [comment]
Running multiple processes in parallel results in each process getting less CPU time. Forking and `exec()`uting a new Python process is resource intensive, so `timeout=5` might be too low for you case; try to increase that.
Also consider using `os.fork()` instead of `subprocess` or use `threading` as the Twisted reactor supports handling multiple connections to different VNC servers at the same time already.

Other than that no issue is known. I'm closing this issue therefore. If you still encounter that issue, please feel free to re-open this issue or to create a new issue.
